.github/workflows/test.yml

```diff
@@ -137,7 +137,7 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, windows-latest]
-        python-version: [3.8, 3.9]
+        python-version: ['3.8', '3.9', '3.10']
```

Comment:
I think we still want to support users on python 3.8, at least for now. Python 3.8's EOL isn't for another 2 years. We dropped python 3.7 in https://github.com/unitaryfund/mitiq/pull/1505 as
1. It's EOL is in 9 months (a little preemptive of us, but also pt 2/3)
2. 3.8 is very similar to 3.7
3. [Numpy 1.22](https://numpy.org/doc/stable/release/1.22.0-notes.html) drops support for python 3.7

Since we are hoping to be on numpy1.22 soon, it just made sense to drop 3.7.

I don't think we have as much reason to drop 3.8 just yet. Are you saying that despite github actions supposedly running in parallel, in effective it never quite does? I've definitely jobs get started at different times as well. Do you think removing 3.8 would make CI builds faster?

Reply:
We do have https://github.com/unitaryfund/mitiq/issues/1427 open to help reduce CI build times, and I agree that dropping 3.8 could help. But I worry that we could break something for python 3.8 users over the next two years. Do you think it's worth the tradeoff?

Reply:
I agree we should continue supporting 3.8 as long as it's practical, so let's keep testing it for now. We can revisit after the changes proposed in #1427, if further reductions in build time become a priority.